core/services/relay/evm/mercury/queue_test.go

```diff
@@ -89,6 +92,11 @@ func Test_Queue(t *testing.T) {
 	})
 
 	t.Run("transmit queue is full and evicts the oldest transmission", func(t *testing.T) {
+		// There is a bug with queue eviction where the evicted transmission is NOT the oldest.
+		// TODO: Once this bug is fixed, replace the expectation with the one below.
```

Comment:
Can it be fixed in this PR?

Reply:
I did try to look for a fix, but I think we need to use a min-max heap or similar data structure and pull in a 3rd party dependency for that. I didn't find any super promising packages (most had 0 or very few stars and no activity in years) from a cursory search so I think some time is needed to find and vet something.

Since this bug is orthogonal to this PR, I was planning to look into it more deeply afterwards. Sergei separately made a ticket for it so I'll check in with him as well if he was planning to take this on.

Reply:
Yup followup is good.

Reply:
I'll own the bug fix and take care of this todo when ready

Reply:
Can we include a link to the ticket in the TODO comment?